carbonmark/components/pages/Project/Retire/Pool/RetireForm.tsx

```diff
@@ -258,6 +268,21 @@ export const RetireForm: FC<Props> = (props) => {
     }
   };
 
+  useEffect(() => {
+    const retirementAddress =
+      inputValues?.beneficiaryAddress ?? (address || "");
+
+    if (retirementBlockNumber !== 0 && retirementIndex) {
+      prepareNavigateToCertificate(
+        router,
+        retirementAddress,
+        retirementIndex,
+        retirementBlockNumber,
+        setSubgraphIndexed
```

Comment:
instead of a callback, just return one of the strings from the uniion `"pending" | "complete" | "timeout"`

```
const handleIndexingComplete = async () => {
  // const status = await prepareNavigateToCertificate();
  const status = await waitForIndexStatus({ blockNumber });
  prefetch(url);
  setIndexStatus(status);
}
```